addon/macros/intl.js

```diff
@@ -0,0 +1,22 @@
+import { computed, get, defineProperty } from '@ember/object';
+import { getOwner } from '@ember/application';
+
+export const __intlInjectionName = `intl-${Date.now().toString(36)}`;
```

Comment:
We have to use a string instead of a symbol here, so that we can add `${__intlInjectionName}.locale` as a dependent key in L21.